common/fee/models.go

```diff
@@ -0,0 +1,102 @@
+package fee
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/common/chains/label"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+var (
+	ErrBumpFeeExceedsLimit = errors.New("fee bump exceeds limit")
+	ErrBump                = errors.New("fee bump failed")
+	ErrConnectivity        = errors.New("transaction propagation issue: transactions are not being mined")
+)
+
+func IsBumpErr(err error) bool {
+	return err != nil && (errors.Is(err, ErrBumpFeeExceedsLimit) || errors.Is(err, ErrBump) || errors.Is(err, ErrConnectivity))
+}
+
+// CalculateFee computes the fee price and chain specific fee limit for a transaction.
+func CalculateFee(
+	feeLimit uint32,
+	maxFeePrice, defaultPrice, maxBumpPrice *big.Int,
+	bumpLimitMultiplier float32,
+) (feePrice *big.Int, chainSpecificFeeLimit uint32, err error) {
+	feePrice, chainSpecificFeeLimit = CapFeePrice(defaultPrice, maxFeePrice, maxBumpPrice, feeLimit, bumpLimitMultiplier)
+	return
+}
+
+// CalculateBumpedFee will increase the price and apply multiplier to the fee limit.
+func CalculateBumpedFee(
+	lggr logger.SugaredLogger,
+	currentFeePrice, originalFeePrice, maxFeePrice,
+	maxBumpPrice, bumpMin *big.Int,
+	originalFeeLimit uint32,
+	bumpPercent uint16,
+	bumpLimitMultiplier float32,
+	chain string,
+) (*big.Int, uint32, error) {
+	feePrice, err := bumpFeePrice(lggr, currentFeePrice, originalFeePrice, maxFeePrice, maxBumpPrice, bumpMin, bumpPercent, chain)
+	if err != nil {
+		return nil, 0, err
+	}
+	chainSpecificFeeLimit := ApplyMultiplier(originalFeeLimit, bumpLimitMultiplier)
```

Comment:
that can also reduce function parameters as well, which is getting pretty long and can be hard to read.

Reply:
Yeah if some params were shared then it would make more sense to me. Maybe an earlier version just passed a config with all of these bundled? :shrug: